integ-tests/openapi/baml_client/openapi.yaml

```diff
@@ -1366,11 +1366,16 @@ components:
             properties:
               email:
                 type: string
+              reason:
+                oneOf:
+                - type: string
+                - type: string
```

Comment:
still need to implement union flattening

Reply:
So I wasn't there for that conversation, so maybe I'm missing something, but here's the result of changing `Resume.name` from `type: string` to `oneOf: [{type: string}, {type: string}]` in golang:

`type: string`:

```
type Resume struct {
        Name string `json:"name"`
        Email string `json:"email"`
        Phone string `json:"phone"`
        Experience []Education `json:"experience"`
        Education []string `json:"education"`
        Skills []string `json:"skills"`
}
```

`oneOf: [{type: string}, {type: string}]`:

```
type Resume struct {
        Name ResumeName `json:"name"`
        Email string `json:"email"`
        Phone string `json:"phone"`
        Experience []Education `json:"experience"`
        Education []string `json:"education"`
        Skills []string `json:"skills"`
}

// ResumeName - struct for ResumeName
type ResumeName struct {
        String *string
}
```

so accessing `resume_obj.Name` now becomes `*resume_obj.Name.string` (and in the case of Go,  you don't even get a guarantee that the string is non-nil!) - this is something we're OK with?

Reply:
yea we're ok with this for now.

The way to resolve this is to resolve literals in the inner layer to a: `Literal['stringA', 'stringB']` isntead of `Literal['stringA'] | Literal['stringB']`.

The reason is type ordering can matter and may cause subtle bugs.

curious, are we sure literals shouldn't be enums? We agreed here its just not as clean? Should we include some docs or something?

Reply:
> The reason is type ordering can matter and may cause subtle bugs.

Do you have any concrete example where that is the case?

> curious, are we sure literals shouldn't be enums? We agreed here its just not as clean?

Unions of string literals can be represented as enums, it adds some complexity but most of the code is already there. I don't think base types are cleaner than enums, neither base types or enums are a one to one representation of literals, so it probably doesn't matter which one we choose in terms of how "clean" it is.

> Should we include some docs or something?

Yes, I can write the docs in a separate PR.